src/atopile/buildutil.py

```diff
@@ -108,10 +109,17 @@ def build(build_ctx: BuildContext, app: Module) -> None:
     except PickError as ex:
         raise UserPickError.from_pick_error(ex) from ex
 
+    # Re-attach because picking might have added new footprints
+    # Many nodes gain their footprints from picking, meaning we'll gather more now
+    transformer.attach(check_unattached=True)
```

Comment:
This only makes sense to do immediately after picking IMO, else anything downstream is liable to use it